File: LatinReader/app/src/main/java/com/ericmschmidt/latinreader/datamodel/Manifest.java
package com.ericmschmidt.latinreader.datamodel;

import com.ericmschmidt.latinreader.MyApplication;

import java.lang.reflect.Constructor;
import java.util.ArrayList;

/** Stores information about the works (source works, translations) in this app.
 * @author Eric Schmidt
 * @author http://telpirion.com
 * @version 1.5
 * @since 1.1
 */
public class Manifest {

    public Manifest() {}
    public ArrayList<WorkInfo> getCollection() {
        return null;
    }
    public WorkInfo getDictionaryInfo() {
        return null;
    }
    public int getDictionaryEntryResource() {
        return 0;
    }

    /**
     * Get the manifest class out of the package using reflection.
     * @param className the fully-qualified class name as a string.
     * @return Manifest
     */
    public static Manifest getManifest(String className) {
        Manifest manifest = null;
        try {
            Class<?> manifestClass = Class.forName(className);
            Constructor<?>[] constructors = manifestClass.getConstructors();
            manifest = (Manifest) constructors[0].newInstance();
        } catch (Exception ex) {
            MyApplication.logError(Manifest.class, ex.getMessage());
        }
        return manifest;
    }
}

File: LatinReader/core-data/src/main/java/com/telpirion/classicsreader/core_data/datamodel/Manifest.kt
package com.telpirion.classicsreader.core_data.datamodel

import java.lang.reflect.InvocationTargetException

/**
 * Stores information about the works (source works, translations) in this app.
 * @author Eric Schmidt
 * @author <a href="http://telpirion.com">...</a>
 * @version 1.5
 * @since 1.1
 */
open class Manifest {

    open fun getCollection(): ArrayList<WorkInfo>? {
        return null
    }

    open fun getDictionaryInfo(): WorkInfo? {
        return null
    }

    open fun getDictionaryEntryResource(): Int {
        return 0
    }

    companion object {
        /**
         * Get the manifest class out of the package using reflection.
         * @param className the fully-qualified class name as a string.
         * @return Manifest
         */
        @Throws(
            ClassNotFoundException::class,
            InvocationTargetException::class,
            IllegalAccessException::class,
            InstantiationException::class
        )
        fun getManifest(className: String): Manifest {
            return try {
                val manifestClass = Class.forName(className)
                val constructor = manifestClass.constructors.first()
                constructor.newInstance() as Manifest
            } catch (ex: Exception) {
                println("Manifest: ${ex.message}")
                throw ex
            }
        }
    }
}